sdk/core/azure-core/test/ut/nullable_test.cpp

```diff
@@ -99,6 +99,34 @@ TEST(Nullable, ValueAssignment)
   EXPECT_FALSE(strVal.HasValue());
 }
 
+TEST(Nullable, Emplace)
+{
+  {
+    Azure::Nullable<int> x;
+    EXPECT_FALSE(x.HasValue());
+    EXPECT_EQ(x.Emplace(42), 42);
+    EXPECT_TRUE(x.HasValue());
+    EXPECT_EQ(x.Value(), 42);
+  }
+
+  {
+    Azure::Nullable<int> y;
+    EXPECT_FALSE(y.HasValue());
+    EXPECT_EQ(y.Emplace(), 0);
+    EXPECT_TRUE(y.HasValue());
+    EXPECT_EQ(y.Value(), 0);
+  }
+
+  {
+    Azure::Nullable<int> z = 42;
+    EXPECT_TRUE(z.HasValue());
+    EXPECT_EQ(z.Value(), 42);
+    EXPECT_EQ(z.Emplace(), 0);
+    EXPECT_TRUE(z.HasValue());
+    EXPECT_EQ(z.Value(), 0);
```

Comment:
Is there any scenario where a `Nullable<T>` that has a value assigned to it (i.e. `HasValue()` is true) changes to be value-less, i.e. `HasValue()` returns false?

Reply:
There is no syntax to emplace with lack-of-value, not even the parameterless `emplace()`.

Reply:
I suspect that if you assign a Nullable<T> with a default constructed Nullable<T>, I believe it will change to valueless.

But that's a strange behavior in the context of nullable.

Reply:
`Reset()` has the semantics I was looking for (both in our `Nullable<T>` and `std::optional`), which sets `HasValue` to false.